### DISCUSSION TITLE
clicking methods to select/deselect curves to see on graph

### DISCUSSION THREAD

**Discussion by ouinouin** [grafana/grafana#54362] [Dashboards, Panels & Visualizations]
Hi, 
I find the selection /deselection method is not easy on grafana since you cannot select deselect whatever curve you want on the graph only with the mouse you have to click on control wich is not very intuitive (one click to enable, ctrl + click to disable and one more click on whatever curve legend is toggling back all the curves . this has another drawback : it makes completely impossible to select/unselect a curve on a mobile/tablet (i know this is absolutely not a target for grafana team but the following proposition solves all in once).
for me the pie widget selection/deselection of the "portions" of the pie is working as expected  : when you click on the legend once, it deselects this portion of the pie and redraws the pie with the correct proportions, when you click again on the same item on the legend, it selects again this portion, clicking many times on the same portion legen doesnt have an effect on the displaying of other portion, wich is the expected behavior.
if all the curves selection/deslection scheme could be done as the pie , it would be more simple. Just remains to find a way to select/unselect/reverse selection (yes reversing the selection would be a nice feat) and I would propose for that a combo with the CRTL key since this looks more an "advanced" action.

I de be glad to have some views on this point, this would make selection : easier , more consistent across the different widgets and add a tiny fraction of usability on tablets and phones (despite the latest can be seen as a side effect : on a tablet you cannot hold ctrl and touch, so you have absolutely no way on a graph to select 2 curves out of 6 for example).

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). Due to the age and number of responses to this discussion, we are deciding to close it. If this is something you would like to see in Grafana, feel free to open an [issue](https://github.com/grafana/grafana/issues) so the discussion can continue. Thank you!
